lib/charms/layer/openstack.py

```diff
@@ -110,15 +111,35 @@ def _save_creds(creds_data):
     else:
         ca_cert = None
 
-    kv().set('charm.openstack.full-creds', dict(
+    value=dict(
         auth_url=endpoint,
+        region=region,
         username=attrs['username'],
         password=attrs['password'],
         user_domain_name=attrs['user-domain-name'],
         project_domain_name=attrs['project-domain-name'],
         project_name=attrs.get('project-name', attrs.get('tenant-name')),
-        endpoint_tls_ca=ca_cert,
-    ))
+        endpoint_tls_ca=ca_cert
+    )        
+    
+    # It only makes sense to check for LBaaS-related information
+    # if we have credentials set up.
+    
+    # Now we look for LBaaS configs
+    if config['subnet-id']:
+        # It only makes sense to advance with lbaas config if both
+        # parameters are set up
+        value['subnet_id'] = config['subnet-id']
+        if config['floating-network-id']:
+            value['fip_id'] = config['floating-network-id']
+        if config['lb-method']:
+            value['lb_method'] = config['lb-method']
+        if config['manage-security-groups']:
```

Comment:
These conditionals are going to cause it to fail when passed into the `set_lbaas_config` function.